### PR TITLE
[Updater] Remove an unused ApiClient method, Make Dependabot::Job an explicit receiver for more parameters

### PR DESCRIPTION
**Pull request by brrygrdn**
Follows up on https://github.com/dependabot/dependabot-core/pull/6770, https://github.com/dependabot/dependabot-core/pull/6792 and https://github.com/dependabot/dependabot-core/pull/6808

This is another clear-the-way PR for #6663 [[Prototype] Generating grouped update PRs](https://github.com/dependabot/dependabot-core/pull/6663).

The primary changes in this PR are:
- Remove ApiClient.fetch_jobs
  - This endpoint was maintained for Dependabot Preview, all current workflows, including the CLI, retrieve this information _outside_ the Updater's container and inject it into a JSON file
- Consolidate all the logic to filter the Job JSON file into Dependabot::Job itself
- Move `repo_contents_path` into Dependabot::Job so we aren't mixing methods of referring back to `Dependabot::Environment` or passing it as an additional argument

The aim of this is to have as little repetition around creating a `Dependabot::Job` as possible as well as binding all inputs _shared_ by the `FileFetcherCommand` and `UpateFilesCommand` to it.

This leaves the `base_commit_sha` and `dependency_files` as floating parameters which are **produced** by the `FileFetcherCommand` and **consumed** by the `UpateFilesCommand`. I don't think it makes sense to bind these to Dependabot::Job as they are a separate concern for that reason. I may bind these into a value class in a subsequent PR if this makes sense.

### CODE REVIEW

**Review by landongrindheim — APPROVED** [review]
I love the direction these changes are moving us toward! 👨‍🍳👌

File: grouped-ruby-job.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
```
[the remark this answers is not in the document]
Comment (brrygrdn) [reply grouped-ruby-job.yml:1]: I did not! Will yoink!

File: updater/lib/dependabot/file_fetcher_command.rb [hunk 2]
```diff
@@ -105,10 +98,18 @@ def file_fetcher
...
-        new(**args)
+      # This bypasses the `job.repo_contents_path` presenter to ensure we fetch
+      # from the file system if the repository contents are mounted even if
+      # cloning is disabled.
+      args[:repo_contents_path] = Environment.repo_contents_path if job.clone? || already_cloned?
+      @file_fetcher ||= Dependabot::FileFetchers.for_package_manager(job.package_manager).new(**args)
+    end
+
+    def already_cloned?
+      return unless Environment.repo_contents_path
```
[the remark this answers is not in the document]
Comment (brrygrdn) [reply updater/lib/dependabot/file_fetcher_command.rb:109]: Ah yes, good catch!

File: updater/lib/dependabot/job.rb [hunk 3]
```diff
@@ -17,11 +17,58 @@
...
+      attrs = standardise_keys(job_definition["job"]).slice(*PERMITTED_KEYS)
+
+      new(attrs.merge(id: job_id, repo_contents_path: repo_contents_path))
+    end
+
+    def self.new_update_job(job_id:, job_definition:, repo_contents_path: nil)
+      attrs = standardise_keys(job_definition["job"]).slice(*PERMITTED_KEYS)
+      # The Updater should only have metadata tokens which allows the proxy to
+      # inject the real credentials as a man-in-the-middle
+      attrs[:credentials] = job_definition.dig("job", "credentials_metadata") || []
```
[the remark this answers is not in the document]
Comment (brrygrdn) [reply updater/lib/dependabot/job.rb:64]: Yep, that reads much better, thanks! ( I am a fan of "<short statement of problem>. Let's do X instead" comments structure in general, can't put my finger on exactly why I always register it better. )

File: updater/lib/dependabot/file_fetcher_command.rb [hunk 4]
```diff
@@ -81,20 +83,11 @@ def base64_dependency_files
...
-          :requirements_update_strategy, :security_advisories,
-          :vendor_dependencies, :experiments, :reject_external_code,
-          :commit_message_options, :security_updates_only
-        )
-
-      @job ||= Job.new(attrs.merge(id: job_id))
+      @job ||= Job.new_fetch_job(
+        job_id: job_id,
+        job_definition: Environment.job_definition,
+        repo_contents_path: Environment.repo_contents_path
```
[the remark this answers is not in the document]
Comment (brrygrdn) [reply updater/lib/dependabot/file_fetcher_command.rb:89]: Yeah, it is currently an Updater-isim but I think we should probably split it out into Core and have an `Dependabot::UpdaterEnvironment` module which extends `Dependabot::Environment` at some point as I like the pattern.

**Pull request merged**